From bd6bed5f6a85e6931de1407f5d9f0f6e3e39573d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sat, 22 May 2021 22:42:36 +0300
Subject: [PATCH] [SW-43025] fwupd: secure i2c update

When an image is sent by I2C we do the same actions in terms of security
as if it was from PCIe, thus the I2C image gets exposed to the PCIe.
As part of the separation, introduce another state field which is
oriented to prevent this exposure of the image.

Change-Id: I6723f6e7dd2c51d7a0b6f6deaf80005c3bf85ae8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index afac8d0..848ffbf 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -42,6 +42,7 @@
 	void (*sts_writel)(u32 val);
 	u32 (*cmd_readl)(void);
 	bool is_disabled;
+	bool use_security;
 };
 
 static struct fwupd_state state;
@@ -245,6 +246,7 @@
 {
 	state.cmd_readl = fwupd_cmd_readl;
 	state.sts_writel = fwupd_sts_writel;
+	state.use_security = true;
 	module_fwupd_send_status(&state, HL_UPD_STS_RST_OK, NULL);
 }
 
@@ -252,6 +254,7 @@
 {
 	i2c_state.cmd_readl = i2c_fwupd_cmd_readl;
 	i2c_state.sts_writel = i2c_fwupd_sts_writel;
+	state.use_security = false;
 	module_fwupd_send_status(&i2c_state, HL_UPD_STS_RST_OK, NULL);
 }
 
@@ -386,10 +389,12 @@
 	state->map_offset = (void *)base_offset;
 	state->map_size = size;
 
-	if (cmd == HL_UPD_MAP_SRAM)
-		preboot_security_dynamic_img_sram_open();
-	else
-		preboot_security_dynamic_img_spsram_open();
+	if (state->use_security) {
+		if (cmd == HL_UPD_MAP_SRAM)
+			preboot_security_dynamic_img_sram_open();
+		else
+			preboot_security_dynamic_img_spsram_open();
+	}
 
 	return sts;
 }
